### PR TITLE
HHH-13076 - Hibernate “Transaction already active” behaviour with custom transaction manager - 5.2

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-13076

This is the PR version of https://github.com/hibernate/hibernate-orm/pull/2630 for 5.2 branch.

The main difference is related to the use of `SessionFactoryOptions#isJpaBootstrap()` instead of the `JpaCompliance.isJpaTransactionComplianceEnabled()`, this setting was introduced in 5.3.

### CODE REVIEW

File: hibernate-core/src/test/java/org/hibernate/test/resource/transaction/jdbc/AlreadyStartedTransactionTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
```
Comment (gsmet) [inline hibernate-core/src/test/java/org/hibernate/test/resource/transaction/jdbc/AlreadyStartedTransactionTest.java]: Maybe one license is enough? :)

File: hibernate-core/src/test/java/org/hibernate/test/resource/transaction/jta/AlreadyStartedTransactionTest.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
```
Comment (gsmet) [inline hibernate-core/src/test/java/org/hibernate/test/resource/transaction/jta/AlreadyStartedTransactionTest.java]: Same here.

**Comment by dreab8** [comment]
@sebersole do u think we can merge this into 5.2?

**Comment by gsmet** [comment]
In the end, we decided against backporting it as the issue is there since 5.2.0 and it's a bit too late to introduce behavior changes.

People having this issue should upgrade to 5.4.x.

**Pull request closed without merging**